### PR TITLE
Chore: Few`ty` fixes

### PR DESCRIPTION
**Pull request by albertkol**
## 📝 Description
Make necessary fixes to make the pipeline pass after upgrading `ty`.

## 🧪 Testing
Are test passing?
Are pipeline passing?

### Review Readiness
- [ ] PR title and description provide sufficient context for reviewers
- [ ] Commits are logical, self-contained, and have clear descriptions

### Performance and security
- [ ] No N+1 query problems introduced
- [ ] Any new DB queries include appropriate where clauses based on the user's permissions

### Testing
- [ ] I have tested this change and it meets the acceptance criteria for the ticket
- [ ] I need the reviewer(s) to pull and run this change locally
- [ ] New (non-developer) functionality has appropriate unit and integration tests
- [ ] End-to-end tests have been updated (if applicable)
- [ ] Edge cases and error conditions are tested

### CODE REVIEW

File: app/common/expressions/registry.py [hunk 1]
```diff
@@ -6,13 +6,15 @@
...
 from collections import defaultdict
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, TypeVar
 
 from app.common.data.types import ManagedExpressionsEnum, QuestionDataType
 
 if TYPE_CHECKING:
     from app.common.expressions.managed import ManagedExpression
 
+TManagedExpression = TypeVar("TManagedExpression", bound="ManagedExpression")
```
Comment (samuelhwilliams) [inline app/common/expressions/registry.py:16]: What's this added for?

File: app/deliver_grant_funding/routes/collections.py [hunk 2]
```diff
@@ -3452,7 +3446,7 @@ def upload_data_set(grant_id: UUID, collection_type: CollectionType, collection_
             data_columns=data_columns,
             preview_data=preview_data,
             s3_key=s3_key,
-            original_filename=secure_filename(file.filename),
+            original_filename=secure_filename(file.filename),  # ty: ignore[invalid-argument-type]
```
Comment (samuelhwilliams) [inline app/deliver_grant_funding/routes/collections.py:3449]: bug: this is covering up an actual typing incompatibility, `file.filename` can be None.

File: app/services/s3.py [hunk 3]
```diff
@@ -39,4 +39,4 @@ def delete_prefix(self, prefix: str) -> None:
 
     def update_file_tags(self, key: str, tags: dict[str, str]) -> None:
         tag_set = [{"Key": k, "Value": v} for k, v in tags.items()]
-        self._client.put_object_tagging(Bucket=self._bucket_name, Key=key, Tagging={"TagSet": tag_set})
+        self._client.put_object_tagging(Bucket=self._bucket_name, Key=key, Tagging={"TagSet": tag_set})  # ty: ignore[invalid-argument-type]
```
Comment (samuelhwilliams) [inline app/services/s3.py:42]: nit: feels like we should fix this, not ignore it, if this is wrong?
